stages/org.osbuild.shell.init

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/python3
+"""
+Write systemwide initialization files, executed for login shells.
+
+Write files to /etc/profile.d with key-value pairs to be set in the shell environment.
+Each file written is sourced by the login shell on initialization.
+"""
+import sys
+import os
+
+import osbuild.api
+
+
+SCHEMA_2 = r"""
+"options": {
+  "properties": {
+    "patternProperties": {
+      "^[\\w.-_]{1,250}$": {
+        "type": "object",
+        "additionalProperties": false,
+        "properties": {
+          "env": {
+            "type": "array",
+            "description": "Environment variables to set on shell init.",
+            "minItems": 1,
+            "items": {
+              "type": "object",
+              "additionalProperties": false,
+              "required": ["key", "value"],
+              "properties": {
+                "key": "string",
+                "value": "string"
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+"""
+
+
+def main(tree, options):
+    for filename, opts in options.items():
+        lines = []
+        for item in opts["env"]:
+            key = item["key"]
+            value = item["value"]
+            lines.append(f'export {key}="{value}"')
+
+        path = os.path.join(tree, "etc/profile.d", filename)
+        with open(f"{path}", "w", encoding="utf8") as f:
```

Comment:
nitpick: you could just write:

```python
with open(f"{tree}/etc/profile.d/{filename}", "w", encoding="utf8") as f:
```
Or
```python
with open(path, "w", encoding="utf8") as f:
```
There is no reason to use the `f""` string with just one variable which is a string.

Reply:
oops.  Yeah, I had the first one then pulled it out to make the path and stupidly kept it as an f-string.

Reply:
Fixed.